appinventor/components/src/com/google/appinventor/components/runtime/SingleValueSensor.java

```diff
@@ -0,0 +1,190 @@
+    } else {
+      sensorManager.registerListener(this, sensor, SensorManager.SENSOR_DELAY_UI);
+    }
+  }
+
+  protected void stopListening() {
+    sensorManager.unregisterListener(this);
+  }
+
+  /**
+   * Available property getter method (read-only property).
+   *
+   * @return {@code true} indicates that a hygrometer is available,
+   *         {@code false} that it isn't
+   */
+  @SimpleProperty
+  public boolean Available() {
+    return isAvailable();
+  }
+
+/**
+   * If true, the sensor will generate events.  Otherwise, no events
+   * are generated.
+   *
+   * @return {@code true} indicates that the sensor generates events,
+   *         {@code false} that it doesn't
+   */
+  @SimpleProperty
+  public boolean Enabled() {
+    return enabled;
+  }
+
+  /**
+   * Specifies whether the sensor should generate events.  If true,
+   * the sensor will generate events.  Otherwise, no events are
+   * generated.
+   *
+   * @param enabled  {@code true} enables sensor event generation,
+   *                 {@code false} disables it
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+      defaultValue = "True")
+  @SimpleProperty
+  public void Enabled(boolean enabled) {
+    setEnabled(enabled);
+  }
+
+  /**
+   * RefreshTime property getter method.
+   *
+   * @return time in ms between updates
+   */
+  @SimpleProperty(
+      description = "The requested minimum time in milliseconds between " +
+      "changed values being reported. This is only a request, which the " +
+      "Android system is not guaranteed to respect.")
+  public int RefreshTime() {
+    return refreshTime;
+  }
+
+  /**
+   * RefreshTime property setter method.
+   *
+   * @param time in ms between updates
+   */
+  @DesignerProperty(
+      editorType = PropertyTypeConstants.PROPERTY_TYPE_NON_NEGATIVE_INTEGER,
+      defaultValue = DEFAULT_REFRESH_TIME + "")
+  @SimpleProperty
+  public void RefreshTime(int time) {
+    refreshTime = time;
+    if (enabled) {
+      stopListening();
+      startListening();
+    }
+  }
+
+  @Override
+  public void onSensorChanged(SensorEvent sensorEvent) {
+    if (enabled && sensorEvent.sensor.getType() == sensorType) {
+      final float[] values = sensorEvent.values;
+      value = values[0];
+      onValueChanged(value);
+    }
+  }
+
+  protected abstract void onValueChanged(float value);
+
+  protected boolean isAvailable() {
+    return sensorManager.getSensorList(sensorType).size() > 0;
+  }
+
+  protected void setEnabled(boolean enabled) {
+    if (this.enabled == enabled) {
+      return;
+    }
+    this.enabled = enabled;
+    if (enabled) {
+      startListening();
+    } else {
+      stopListening();
+    }
+  }
+
+  @Override
+  public void onAccuracyChanged(Sensor sensor, int accuracy) {}
+    
+  @Override
+  public void onResume() {
+    if (enabled) {
+      startListening();
+    }
+  }
+
+  @Override
+  public void onStop() {
```

Comment:
Thanks for catching that. That's a big deal!

I modeled the code off Accelerometer.java, which is also missing onPause(). Should I fix that in this PR or another one?

Reply:
It's not clear to me that we need onStop() if we have onPause(), since I believe all calls to onStop() are preceded by onPause(). Do you have an opinion on this?

Reply:
Based on Figure 1 from [Understanding the Activity Lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle?hl=en), onPause should always be called before onStop, so I agree that we should just use onPause instead.